src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -462,22 +453,22 @@ public long getVulnerabilityCount(Project project, boolean includeSuppressed) {
      * @return a List of Vulnerability objects
      */
     public List<Vulnerability> getVulnerabilities(Project project, boolean includeSuppressed) {
-        final List<Vulnerability> vulnerabilities = new ArrayList<>();
-        final List<Component> components = getAllComponents(project);
-        for (final Component component: components) {
-            final Collection<Vulnerability> componentVulns = pm.detachCopyAll(
-                    getAllVulnerabilities(component, includeSuppressed)
-            );
-            Map<String, Epss> matchedEpssList = getEpssForCveIds(
-                    componentVulns.stream().map(vuln -> vuln.getVulnId()).distinct().toList());
-            for (final Vulnerability componentVuln: componentVulns) {
-                componentVuln.setComponents(Collections.singletonList(pm.detachCopy(component)));
-                componentVuln.setAliases(new ArrayList<>(pm.detachCopyAll(getVulnerabilityAliases(componentVuln))));
-                componentVuln.setEpss(matchedEpssList.get(componentVuln.getVulnId()));
+        List<Vulnerability> projectVulnerabilities;
+        var vulnerableComponents = new HashMap<Long, Component>();
+        try (final Handle jdbiHandle = openJdbiHandle((this.request))) {
+            final var dao = jdbiHandle.attach(VulnerabilityDao.class);
+            projectVulnerabilities = dao.getVulnerabilitiesByProject(project.getId(), includeSuppressed);
+            dao.getVulnerableComponents(project.getId(), projectVulnerabilities.stream().map(Vulnerability::getId).toList())
+                    .stream().forEach(vc -> vulnerableComponents.put(vc.getId(), vc));
+        }
+        vulnerableComponents.values().forEach(vc -> vc.setProject(project));
```

Comment:
Seeing as the project information is repeated in the API response for every component, I wonder if we should just omit this. I think the inclusion of it was more of a bug in the previous implementation than an intentional inclusion.

Since clients are requesting vulnerabilities for a specific project already, do we really need to include that project's details in the response again?

I do realize this would technically be a breaking change, but I can't fathom anyone actually needing this kind of nested and redundant data. Thoughts?

Reply:
yeah I completely agree, there is no point of nested project info in the component data when it is requested of specific project. I'll remove this part 👍🏼